client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/customs-step/index.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import StepContainer from '../step-container';
+import { hasNonEmptyLeaves } from 'woocommerce/woocommerce-services/lib/utils/tree';
+import { toggleStep, confirmCustoms } from 'woocommerce/woocommerce-services/state/shipping-label/actions';
+import {
+	getShippingLabel,
+	isLoaded,
+	getFormErrors,
+	isCustomsFormStepSubmitted,
+} from 'woocommerce/woocommerce-services/state/shipping-label/selectors';
+import StepConfirmationButton from '../step-confirmation-button';
+import Checkbox from 'woocommerce/woocommerce-services/components/checkbox';
+
+const CustomsStep = ( props ) => {
+	const {
+		errors,
+		expanded,
+		translate,
+		isSubmitted,
+		dummyFieldChecked,
+	} = props;
+	const summary = hasNonEmptyLeaves( errors ) ? translate( 'Customs information incomplete' ) : translate( 'Customs information valid' );
+
+	return (
+		<StepContainer
+			title={ translate( 'Customs' ) }
+			summary={ isSubmitted ? summary : '' }
+			expanded={ expanded }
+			toggleStep={ props.toggleStep }
+			isSuccess={ isSubmitted && ! hasNonEmptyLeaves( errors ) }
+			isError={ isSubmitted && hasNonEmptyLeaves( errors ) } >
+			<Checkbox checked={ Boolean( dummyFieldChecked ) } onChange={ props.toggleDummyField } /> Check this to make validation pass.
+			<StepConfirmationButton
+				disabled={ hasNonEmptyLeaves( errors ) }
+				onClick={ props.confirmCustoms } >
+				{ translate( 'Confirm' ) }
+			</StepConfirmationButton>
+		</StepContainer>
+	);
+};
+
+CustomsStep.propTypes = {
+	siteId: PropTypes.number.isRequired,
+	orderId: PropTypes.number.isRequired,
+};
+
+const mapStateToProps = ( state, { orderId, siteId } ) => {
+	const loaded = isLoaded( state, orderId, siteId );
+	const shippingLabel = getShippingLabel( state, orderId, siteId );
+
+	return {
+		expanded: shippingLabel.form.customs.expanded,
+		dummyFieldChecked: shippingLabel.form.customs.dummyFieldChecked,
+		isSubmitted: isCustomsFormStepSubmitted( state, orderId, siteId ),
+		errors: loaded ? getFormErrors( state, orderId, siteId ).customs : {},
+	};
+};
+
+const mapDispatchToProps = ( dispatch, { orderId, siteId } ) => ( {
+	toggleStep: () => dispatch( toggleStep( orderId, siteId, 'customs' ) ),
```

Comment:
Was there a reason to deviate from the usual `mapDispatchToProps` pattern for the steps:

```
const mapDispatchToProps = dispatch => {
	return bindActionCreators( { toggleStep, confirmPackages }, dispatch );
};
```

Reply:
I'm trying to implement it this way when it makes sense. The reason is that, with this implementation, the component just needs to call `props.toggleStep()`. With a plain `bindActionCreators`, the component would need to call `props.toggleStep( props.orderId, props.siteId, 'customs' )`. I prefer to have abit of logic in the `mapDispatchToProps` so the component can be "dumber".